tests/fixtures/synthetic_hf_state.py

```diff
@@ -0,0 +1,78 @@
+import pytest
+from composer.core.state import State
+from composer.datasets.dataloader import DataLoaderHparams
+from composer.datasets.lm_datasets import LMDatasetHparams
+from composer.models import BERTHparams, GPT2Hparams
+from tests.common.models import generate_dummy_model_config
+from tests.datasets import test_synthetic_lm_data
+from composer.datasets.synthetic_lm import generate_synthetic_tokenizer, synthetic_hf_dataset_builder
+
+def make_dataset_configs():
+    lm_dataset_configs = [config[0] for config in test_synthetic_lm_data.generate_parameter_configs( ['num_samples', 'chars_per_sample', 'column_names', 'tokenizer_family'])]
+    for config in lm_dataset_configs:
+        config['drop_last'] = False
+        config['use_masked_lm'] = config['tokenizer_family'] == 'bert'
+        if config['use_masked_lm']:
+            config['mlm_probability'] = 0.15
+    return lm_dataset_configs
+
+def make_lm_tokenizer(config: dict):
+    dataset = synthetic_hf_dataset_builder(num_samples=config['num_samples'],
+                                        chars_per_sample=config['chars_per_sample'],
+                                        column_names=config['column_names'])
+    tokenizer = generate_synthetic_tokenizer(config['tokenizer_family'], dataset)
+    return tokenizer
+
+def make_dummy_lm(model_name: str, max_position_embeddings: int, tokenizer):
+    pytest.importorskip("transformers")
+    if model_name == 'gpt2':
+        class_name = GPT2Hparams
+    elif model_name == 'bert':
+        class_name = BERTHparams
+    model_config = generate_dummy_model_config(class_name, tokenizer)
+    if model_name == 'bert':
```

Comment:
@jzf2101 Should we just fold these into the `generate_dummy_model_config` function?

Reply:
It looks like I was being overly detailed. [`num_labels` is set when calling `BERTHparams` to `vocab_size`](https://github.com/mosaicml/composer/blob/23f888e07456d31a2950db9f962f4dbdf41938fb/composer/models/bert/bert_hparams.py#L114)

Reply:
This issue is handled in #1023